src/librustc_metadata/encoder.rs

```diff
@@ -282,7 +282,7 @@ fn encode_symbol(ecx: &EncodeContext,
 
 fn encode_disr_val(_: &EncodeContext,
                    rbml_w: &mut Encoder,
-                   disr_val: ty::Disr) {
+                   disr_val: u64) {
```

Comment:
Perhaps it should still take `ty::Disr` here and convert to whatever integer it wants inside.